Blish HUD/Controls/_Types/ControlCollection.cs

```diff
@@ -59,7 +59,8 @@ private class ControlEnumerator<TEnum> : IEnumerator<TEnum> {
         }
 
         public IEnumerator<T> GetEnumerator() {
-            _listLock.EnterReadLock();
+            if (!_listLock.IsReadLockHeld)
```

Comment:
To clarify, is this for same-thread locking because `ReaderWriterLockSlim` does not support reentrants here?

Reply:
Yes when the same thread tries to enter the lock a second time, it throws an exception, preventing following code from executing.

Reply:
Sounds good.  👍